### PR TITLE
[5.2] Update Router::auth() give names to routes 

### PR DESCRIPTION
**Pull request by mohamedsharaf**
public function auth() give names to routes //login , logout , register , passwordEmail , passwordReset

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Your code contains syntax errors.

File: src/Illuminate/Routing/Router.php [hunk 1]
```diff
@@ -346,17 +346,17 @@ public function resource($name, $controller, array $options = [])
...
         // Authentication Routes...
-        $this->get('login', 'Auth\AuthController@showLoginForm');
+        $this->get('login', 'Auth\AuthController@showLoginForm')->name('login');
         $this->post('login', 'Auth\AuthController@login');
-        $this->get('logout', 'Auth\AuthController@logout');
+        $this->get('logout', 'Auth\AuthController@logout')->name('logout');
 
         // Registration Routes...
-        $this->get('register', 'Auth\AuthController@showRegistrationForm');
+        $this->get('register', 'Auth\AuthController@showRegistrationForm')->('register');
```
Comment (GrahamCampbell) [inline src/Illuminate/Routing/Router.php]: syntax error here

**Comment by mohamedsharaf** [comment]
i suggest we add / to the base route also it be like this /login instead login

// Authentication Routes...
        $this->get('/login', 'Auth\AuthController@showLoginForm')->name('login');

**Comment by mohamedsharaf** [comment]
done

File: src/Illuminate/Routing/Router.php [hunk 2]
```diff
@@ -346,18 +346,18 @@ public function resource($name, $controller, array $options = [])
     public function auth()
     {
         // Authentication Routes...
-        $this->get('login', 'Auth\AuthController@showLoginForm');
-        $this->post('login', 'Auth\AuthController@login');
-        $this->get('logout', 'Auth\AuthController@logout');
+        $this->get('/login', 'Auth\AuthController@showLoginForm')->name('login');
```
Comment (garygreen) [inline src/Illuminate/Routing/Router.php]: Why adding forward slashes?

**Comment by garygreen** [comment]
I would prefer to have these namespaced like `auth.login` `auth.logout` `auth.register` `auth.reset-password`

**Comment by mohamedsharaf** [comment]
if you need to prefix it using group prefix it will make sense 
Route::group(['prefix' => '/user' , 'middleware' => ['web']], function () {
Route::auth();
}

may be /locale /user ,/member



also names you suggested is good but this depending on laravel owner Taylor which names they prefer

**Comment by garygreen** [comment]
@mohamedsharaf true about the group. That's probably a better solution.

**Comment by taylorotwell** [comment]
Holding off on this simply because everyone will have different opinions on what the names should be. If you want names you can copy and paste the routes into your own routes file and name them whatever you want.

**Comment by garygreen** [comment]
I think it makes sense to do it by default though -- it sets a good example and good practise. If the names aren't quite to your liking *then* define your own in your routes file.

**Comment by mohamedsharaf** [comment]
@taylorotwell yes we can copy and paste but why not simply copy and there is good function do it 
every one have different opinon  you are right for this i choosed same name as route 
/login name is : login
if not right you just chose the proper one you see 

i just tried to make function better 
/ to make ability be prefix in a group
and the name make sense so u can easily get the rout by name or by helper route()

@garygreen yes it make sense

**Pull request closed without merging**